SecretManagement.LastPass.Extension/SecretManagement.LastPass.Extension.psm1

```diff
@@ -174,51 +202,51 @@ function Set-Secret
         $Keys = $Secret.Keys.Where({$_ -notin $SpecialKeys })
         foreach ($k in $Keys) {
             if ($Secret.$k -is [securestring]) {
-                $BSTR = [System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($Secret.$k)
-                $Secret.$k = [System.Runtime.InteropServices.Marshal]::PtrToStringAuto($BSTR)
+                $Secret.$k = [System.Net.NetworkCredential]::new("", $Secret.$k).Password
             }
             $sb.AppendLine("$($k): $($Secret.$k)") | Out-Null
         }
 
         # Notes need to be on a new line
         if ($null -ne $Secret.Notes) {
             if ($Secret.Notes -is [securestring]) {
-                $BSTR = [System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($Secret.Notes)
-                $Secret.Notes = [System.Runtime.InteropServices.Marshal]::PtrToStringAuto($BSTR)
+                $Secret.Notes = [System.Net.NetworkCredential]::new("", $Secret.Notes).Password
             }
             $sb.AppendLine("Notes: `n$($Secret.Notes)") | Out-Null
         }
     } 
     
-    try {
-        $res = Invoke-lpass 'show', '--sync=now', '--name', $Name, '2>/dev/null' -ErrorAction Stop 
-        $SecretExists = $null -ne $res 
-
-        if ($SecretExists) {
-            Write-Verbose "Editing secret" 
-            $sb.ToString() | Invoke-lpass 'edit', '--non-interactive', $Name
-        } else {
-            Write-Verbose "Adding new secret" 
-            $NoteTypeArgs = @()
-            if ($null -ne $Secret.NoteType) {
-                if ($Secret.NoteType -is [securestring]) { 
-                    $BSTR = [System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($Secret.NoteType)
-                    $Secret.NoteType = [System.Runtime.InteropServices.Marshal]::PtrToStringAuto($BSTR)
-                }
-                $NoteTypeArgs += "--note-type=$($Secret.NoteType)"
+    $res = Invoke-lpass 'show', '--sync=now', '--name', $Name
+
+    # We use ToString() here to turn the ErrorRecord into a string if we got an ErrorRecord
+    $SecretExists = if ($null -eq $res) {
+        # This should never ever happen...
+        Write-Warning "Querying the secret $Name produced an unexpected result of `$Null"
+        $false
+    } elseif ($res.ToString() -eq $lpassMessage.AccountNotFound) {
+        $false
+    } else {
+        $true
+    }
+
+    if ($SecretExists) {
+        Write-Verbose "Editing secret"
+        $sb.ToString() | Invoke-lpass 'edit', '--non-interactive', $Name
```

Comment:
The same kind of problem might occur on the edit operation than the below comment.
If the Edit fail, we would want to notify the user too.

Reply:
My last commit should fix this generically.